src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Learner.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace SFA.DAS.CommitmentsV2.Models
+{
+    // This is a pseudo-entity to represent the result of the GetLearnersBatch stored proc, it's not a table in the database.
```

Comment:
This feels very weird that we have to add a pseudo Learner object to deal with the stored procedure. I can understand the why.

Looking at the stored procedure contents, and what the stored procedure does, I think it would have been easier to do this in Linq, avoiding all the stored procedure calls and mapping, and also this pseudo object.

The GetAllLearnersQueryHandler, could have done a join instead of the stored proc. Then projected the result into the Learner Type, which at that point could have been a simple model, rather than a requirement for a pseudo EF object with configuration.

The linq join could also have used the Skip and Take methods to correspond to the batchsize / batch number. This is something I would like to introduce if we can into this PR but appreciate time constraints may not allow along with reworking the integration tests.

Reply:
Agreed.